src/components/OptionsSelector/BaseOptionsSelector.js

```diff
@@ -61,622 +54,461 @@ const defaultProps = {
-        this.sliceSections = this.sliceSections.bind(this);
-        this.calculateAllVisibleOptionsCount = this.calculateAllVisibleOptionsCount.bind(this);
-        this.handleFocusIn = this.handleFocusIn.bind(this);
-        this.handleFocusOut = this.handleFocusOut.bind(this);
-        this.debouncedUpdateSearchValue = _.debounce(this.updateSearchValue, CONST.TIMING.SEARCH_OPTION_LIST_DEBOUNCE_TIME);
-        this.relatedTarget = null;
-        this.accessibilityRoles = _.values(CONST.ROLE);
-        this.isWebOrDesktop = [CONST.PLATFORM.DESKTOP, CONST.PLATFORM.WEB].includes(getPlatform());
-
-        const allOptions = this.flattenSections();
-        const sections = this.sliceSections();
-        const focusedIndex = this.getInitiallyFocusedIndex(allOptions);
-        this.focusedOption = allOptions[focusedIndex];
-
-        this.state = {
-            sections,
-            allOptions,
-            focusedIndex,
-            shouldDisableRowSelection: false,
-            errorMessage: '',
-            paginationPage: 1,
-            disableEnterShortCut: false,
-            value: '',
-        };
-    }
-
-    componentDidMount() {
-        this.subscribeToEnterShortcut();
-        this.subscribeToCtrlEnterShortcut();
-        this.subscribeActiveElement();
-
-        if (this.props.isFocused && this.props.autoFocus && this.textInput) {
-            this.focusTimeout = setTimeout(() => {
-                this.textInput.focus();
-            }, CONST.ANIMATED_TRANSITION);
-        }
-
-        this.scrollToIndex(this.props.selectedOptions.length ? 0 : this.state.focusedIndex, false);
-    }
+function BaseOptionsSelector(props) {
+    const isFocused = useIsFocused();
+    const {translate} = useLocalize();
+    const themeStyles = useThemeStyles();
 
-    componentDidUpdate(prevProps, prevState) {
-        if (prevState.disableEnterShortCut !== this.state.disableEnterShortCut) {
-            // Unregister the shortcut before registering a new one to avoid lingering shortcut listener
-            this.unsubscribeEnter();
-            if (!this.state.disableEnterShortCut) {
-                this.subscribeToEnterShortcut();
-            }
-        }
+    const [disabledOptionsIndexes, setDisabledOptionsIndexes] = useState([]);
+    const [errorMessage, setErrorMessage] = useState('');
+    const [value, setValue] = useState('');
+    const [paginationPage, setPaginationPage] = useState(1);
 
-        if (prevProps.isFocused !== this.props.isFocused) {
-            // Unregister the shortcut before registering a new one to avoid lingering shortcut listener
-            this.unSubscribeFromKeyboardShortcut();
-            if (this.props.isFocused) {
-                this.subscribeToEnterShortcut();
-                this.subscribeToCtrlEnterShortcut();
-            }
-        }
+    const shouldDisableRowSelection = useRef(false);
+    const relatedTarget = useRef(null);
+    const listRef = useRef();
+    const textInputRef = useRef();
```

Comment:
We prefer to set a default value to `undefined`.
```suggestion
    const listRef = useRef(null);
    const textInputRef = useRef(null);
```